Description: Add support for xmonad (#536847)
 The attached patch adds support for xmonad. Without this
 patch, matchbox-keyboard can be focused like any other window,
 which makes it not useful. With the patch, xmonad does not manage
 it at all.
 .
 It might be better to use the STRUTS property to communicate
 with xmonad, but just uncommenting the code that was already
 there Worked for Me.
Author: Joey Hess <joeyh@debian.org>
Last-Update: 2009-09-13
Index: matchbox-keyboard-0.2+git20160713/src/matchbox-keyboard-ui.c
===================================================================
--- matchbox-keyboard-0.2+git20160713.orig/src/matchbox-keyboard-ui.c	2020-06-15 17:32:16.993221445 +0100
+++ matchbox-keyboard-0.2+git20160713/src/matchbox-keyboard-ui.c	2020-06-15 17:33:59.473195981 +0100
@@ -676,6 +676,7 @@
       char                *wm_name;
       boolean              have_matchbox_wm = False;
       boolean              have_ewmh_wm     = False;
+      boolean              have_xmonad_wm   = False;
       int                  desk_width = 0, desk_height = 0, desk_y = 0;
 
       /*
@@ -733,6 +734,8 @@
 
       if (wm_name && streq(wm_name, "matchbox"))
         have_matchbox_wm = True;
+      else if (wm_name && streq(wm_name, "xmonad"))
+	have_xmonad_wm = True;
 
       win_attr.override_redirect = False; /* Set to true for extreme case */
       win_attr.event_mask =
@@ -865,15 +868,12 @@
                                   (unsigned char *)
                                   &atom_NET_WM_WINDOW_TYPE_TOOLBAR, 1);
                 }
-              else
+              else if (have_xmonad_wm)
                 {
-                  /*
                     XChangeProperty(ui->xdpy, ui->xwin,
                     atom_NET_WM_WINDOW_TYPE, XA_ATOM, 32,
                     PropModeReplace,
                     (unsigned char *) &atom_NET_WM_WINDOW_TYPE_DOCK, 1);
-                  */
-
                 }
             }
         }
